napari/layers/base/_base_mouse_bindings.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+
+from napari.components.overlays._interaction_box_utils import (
+    InteractionBoxHandle,
+    generate_transform_box_from_layer,
+    get_nearby_handle,
+)
+from napari.utils.transforms import Affine
+
+
+def highlight_box_handles(layer, event):
+    if not len(event.dims_displayed) == 2:
+        return
+
+    pos_data = layer.world_to_data(event.position)
+    pos_displayed = np.array(pos_data)[event.dims_displayed]
+
+    handle_coords = generate_transform_box_from_layer(
+        layer, event.dims_displayed
+    )
+    nearby_handle = get_nearby_handle(pos_displayed, handle_coords)
+
+    # set the selected vertex of the box to the nearby_handle (can also be INSIDE or None)
+    layer._overlays['transform_box'].selected_vertex = nearby_handle
+
+
+def transform_with_box(layer, event):
```

Comment:
Probably I shoudln't be working in data coords, but canvas coords... Anyways, I added some comments and renamed some variable, hopefully it's understandable.

Reply:
Aha! I solved some of the issues; I was actually comparing data coordinates from the handles with world coordinates from the mouse. Now most things are correct, except rescaling. Rescaling is still broken because it's resulting in a shear...

Reply:
I was trying to manually do that by applying rotation and translation, but I guess I was overcomplicating and still failing somehow. I'll try that, thanks for the pointer!

Reply:
I got it! :D